activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -164,7 +164,7 @@ def resolve_all
         #   spec.config
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" }
         #
-        def spec(config)
+        def spec(config, id = nil)
```

Comment:
Agree `id` is awkward. To me `name` is slightly better.. I'm struggling to come up with a particularly compelling alternative, though. :confused: